arch/arm/boot/dts/overlays/README

```diff
@@ -3599,6 +3599,13 @@ Load:   dtoverlay=vc4-kms-dpi-at056tn53v1
 Params: <None>
 
 
+Name:   vc4-kms-dpi-geekworm-mzp280
```

Comment:
I was looking at this yesterday. Having independent overlays for each and every display is going to become cumbersome. Adding overrides to vc4-kms-dpi-generic (or create one new one of vc4-kms-dpi for defined panels) would be more manageable.
vc4-kms-dpi-at056tn53v1 can be merged into -generic quite easily.
More complex displays (eg the Pimoroni HyperPixels that need spi-gpio for initialisation, and have touch controllers to configure too) would need to remain as separate overlays.

I wonder what @pelwell 's view is on that.

Reply:
I'm all for avoid unnecessary duplication - whether that's by adding specialisations to the `-generic` case or by adding a `-specific` version for that purpose (name TBD) doesn't matter much, but the latter is marginally more appealing.

Reply:
I'll be happy to remove this request if we can do it in a generic way. Honestly the only 2 (3) things that prevent it from working with the generic overlay today are:
1) There is no bus format defined. That can however be accomplished with 2 of the patches in this series and doesn't require the overlay at all to do. If this is the route you want to take I can resubmit this pull request with just the bus format changes to the VC4 driver and to the media headers.

2) I need to enable SPI0 but keep the SPI0 CS pins in ALT2 mode for the touchscreen to work. If I don't enable SPI0 there is no touchscreen, but if I don't keep the CS pins in ALT2 mode I get a horrible blue tint.

3-ish) This is kind of optional, but I should/could add a rotation property to the DPI generic overlay. The panel I'm submitting for in question is by default in a portrait orientation and is almost always rotated 270 degrees when used.

Let me know what you want me to do and I'll resubmit. Thank you.

Reply:
- panel-simple.c got extended to take a ```bus-format``` property to allow the format to be specified when using the ```panel-dpi``` compatible.
You've gone to the effort of getting a compatible merged upstream (congrats!), so you may as well use it, and that defines the bus format anyway. I'm fine with that. You need the patches adding that RGB565 mode to vc4_dpi.c anyway.

- Enabling SPI isn't done in this overlay, so changing chip selects shouldn't really be done in this overlay either. Which touch controller is it using? Is there an existing overlay for it that needs to be extended?

- I'm all for additional params if they make sense.

- I've had a quick chat with pelwell. Where we will be going will be a ```vc4-kms-dpi-panel``` overlay which takes an override to enable different defined panels (eg yours, and at056tn53v1).
```vc4-kms-dpi-generic``` will exist for providing a generic external timing/configuration.
Both will include a new ```vc4-kms-dpi.dtsi``` which will define the base panel, dpi, and probably backlight properties.

It may take a day or so for me to find time to make the above changes. If you want to resubmit before those changes, then if you create the ```vc4-kms-dpi-panel-overlay.dts``` file with your panel as the only override, then we can do the shuffling of fragments around the place. If we merge your panel as ```vc4-kms-dpi-geekworm-mzp280```, then it becomes an overhead that has to be carried forward to avoid breaking someone's system.